core/storage/question/gae_models_test.py

```diff
@@ -379,14 +387,18 @@ def test_get_more_question_skill_links_than_available(self):
         # Testing for queries that retrieve more questions than available.
         question_skill_links = (
             question_models.QuestionSkillLinkModel.
-            get_question_skill_links_equidistributed_by_skill(
-                4, ['skill_id1', 'skill_id2']
+            get_question_skill_links_based_on_mastery_equidistributed_by_skill(
+                4, ['skill_id1', 'skill_id2'],
+                {'skill_id1': 0.4, 'skill_id2': 0.9}
             )
         )
         self.assertEqual(len(question_skill_links), 3)
         self.assertEqual(question_skill_links[0].skill_id, 'skill_id1')
+        self.assertEqual(question_skill_links[0].question_id, 'question_id2')
         self.assertEqual(question_skill_links[1].skill_id, 'skill_id1')
+        self.assertEqual(question_skill_links[1].question_id, 'question_id1')
         self.assertEqual(question_skill_links[2].skill_id, 'skill_id2')
+        self.assertEqual(question_skill_links[2].question_id, 'question_id3')
 
     def test_get_question_skill_links_when_count_not_evenly_divisible(self):
```

Comment:
Test a case where degrees of mastery is None, and a case where degrees of mastery doesn't have a value for a skill id

Reply:
Done.